src/main/navigation/navigation_fixedwing.c

```diff
@@ -357,6 +383,37 @@ static void updatePositionHeadingController_FW(timeUs_t currentTimeUs, timeDelta
     // We have virtual position target, calculate heading error
     int32_t virtualTargetBearing = calculateBearingToDestination(&virtualDesiredPosition);
 
+    /* If waypoint tracking enabled quickly force craft toward waypoint course line and track along it */
+    if (navConfig()->fw.waypoint_tracking_accuracy && isWaypointNavTrackingRoute() && !needToCalculateCircularLoiter) {
+        // only apply course tracking correction if target bearing error < 90 degs or when close to waypoint (within 10m)
+        if (ABS(wrap_18000(virtualTargetBearing - posControl.actualState.yaw)) < 9000 || posControl.wpDistance < 1000.0f) {
+            fpVector3_t virtualCoursePoint;
+            virtualCoursePoint.x = posControl.activeWaypoint.pos.x -
+                                   posControl.wpDistance * cos_approx(CENTIDEGREES_TO_RADIANS(posControl.activeWaypoint.yaw));
+            virtualCoursePoint.y = posControl.activeWaypoint.pos.y -
+                                   posControl.wpDistance * sin_approx(CENTIDEGREES_TO_RADIANS(posControl.activeWaypoint.yaw));
+            float distToCourseLine = calculateDistanceToDestination(&virtualCoursePoint);
+
+            // courseVirtualCorrection initially used to determine current position relative to course line for later use
+            int32_t courseVirtualCorrection = wrap_18000(posControl.activeWaypoint.yaw - virtualTargetBearing);
+
+            // bias between reducing distance to course line and aligning with course heading adjusted by waypoint_tracking_accuracy
+            float courseCorrectionFactor = constrainf(distToCourseLine / (100.0f * navConfig()->fw.waypoint_tracking_accuracy), 0.0f, 1.0f);
+            courseCorrectionFactor = courseVirtualCorrection < 0 ? -courseCorrectionFactor : courseCorrectionFactor;
+
+            int32_t courseHeadingError = wrap_18000(posControl.activeWaypoint.yaw - posControl.actualState.yaw);
+            float courseHeadingFactor = constrainf(sq(courseHeadingError / 18000.0f), 0.0f, 1.0f);
+            courseHeadingFactor = courseHeadingError < 0 ? -courseHeadingFactor : courseHeadingFactor;
+
+            courseCorrectionFactor = constrainf(courseCorrectionFactor - courseHeadingFactor, -1.0f, 1.0f);
+
+            // final courseVirtualCorrection using max 80 deg heading adjustment toward course line
+            courseVirtualCorrection = 8000 * courseCorrectionFactor;
```

Comment:
can you please move degrees to the config variable? 
for my needs, I more like angle 30 degrees and 60 accuracy with speed 22m/s
![Screenshot from 2022-07-31 18-30-02](https://user-images.githubusercontent.com/4169835/182033722-9cf79db9-614d-4b3c-894d-f2503b66d9f1.png)

Reply:
@sevrugin Didn't reply to your comments because I'm going to change the tracking accuracy to a different method that works on X/Y tracking rather than headings. Seems better and more reliable tuning wise. Not sure how that will work out in terms of adjusting the convergence angle but there is a setting which should do something similar.

Reply:
I also changed the turn smoothing so there's now another option to cut inside the turn without passing through the waypoint which is what you suggested above I think. Not tested it should work since it just shifts the centre of the loiter further inside the turn away from the waypoint.

Reply:
will it be optional what method we want to use for the turns? I am for my purposes not a fan of that method. 
looking forward to the new tracking method.

Reply:
2 options, one passing through the waypoint and the other cutting the turn missing the waypoint. Haven't tested it though so you might find there's not a lot of difference depending on how the plane behaves in the turn.

Reply:
ok good. I am trying to figure out how to fork INAV and merge your PR + the HITL change and maybe other stuff in it and build it myself, so I can test it in the sim first and then in the field. This git stuff is driving me crazy. And currently it is too hot again to go fly. at least till the weekend